src/Shared/SegmentWriteStream.cs

```diff
@@ -99,12 +99,11 @@ protected override void Dispose(bool disposing)
 
     public override void Flush()
     {
-        if (!CanWrite)
-        {
-            throw new ObjectDisposedException(nameof(SegmentWriteStream), "The stream has been closed for writing.");
```

Comment:
This feels a little different - does this only happen after the stream is closed?

Reply:
Personally, I like the exception since I think flushing (or doing anything else) after Dispose is a programming error, but I see that `CryptoStream` itself [doesn't](https://source.dot.net/#System.Security.Cryptography/System/Security/Cryptography/CryptoStream.cs,203) throw under those circumstances, so I'm okay with not throwing here either.